core/server/worker/src/main/java/alluxio/worker/block/BlockStore.java

```diff
@@ -35,12 +35,23 @@
    *
    * @param sessionId the id of the session to lock this block
    * @param blockId the id of the block to lock
-   * @return the lock id if the lock is acquired successfully
+   * @return the lock id (non-negative) if the lock is acquired successfully
    * @throws BlockDoesNotExistException if block id can not be found, for example, evicted already
    */
   long lockBlock(long sessionId, long blockId) throws BlockDoesNotExistException;
 
   /**
+   * Locks an existing block and guards subsequent reads on this block. If the lock fails, return
+   * {@link BlockLockManager#INVALID_LOCK_ID}.
+   *
+   * @param sessionId the id of the session to lock this block
+   * @param blockId the id of the block to lock
+   * @return the lock id (non-negative) that uniquely identifies the lock obtained or
+   *         {@link BlockLockManager#INVALID_LOCK_ID} if it failed to lock
+   */
+  long lockBlockNoException(long sessionId, long blockId);
```

Comment:
no, i was saying that if we still want to throw BlockDoesNotExistException when lock fails, it will be 1) less readable since people would wonder why are we throwing that exception when lockBlock return false. 2) more error-prone. what if the cause of the lock failure cause changes in the future. 3) loses stack trace in the exception.

one way to solve this is to return the cause when lockBlock fails.